KIPs/kip-17-non_fungible_token.md

```diff
@@ -0,0 +1,544 @@
+
+```solidity
+pragma solidity 0.4.24;
+
+/// @title KIP-17 Non-Fungible Token Standard, optional enumeration extension
+///  Note: the KIP-13 identifier for this interface is 0x780e9d63.
+interface KIP17Enumerable {
+    /// @notice Count NFTs tracked by this contract
+    /// @return A count of valid NFTs tracked by this contract, where each one of
+    ///  them has an assigned and queryable owner not equal to the zero address
+    function totalSupply() external view returns (uint256);
+
+    /// @notice Enumerate valid NFTs
+    /// @dev Throws if `_index` >= `totalSupply()`.
+    /// @param _index A counter less than `totalSupply()`
+    /// @return The token identifier for the `_index`th NFT,
+    ///  (sort order not specified)
+    function tokenByIndex(uint256 _index) external view returns (uint256);
+
+    /// @notice Enumerate NFTs assigned to an owner
+    /// @dev Throws if `_index` >= `balanceOf(_owner)` or if
+    ///  `_owner` is the zero address, representing invalid NFTs.
+    /// @param _owner An address where we are interested in NFTs owned by them
+    /// @param _index A counter less than `balanceOf(_owner)`
+    /// @return The token identifier for the `_index`th NFT assigned to `_owner`,
+    ///   (sort order not specified)
+    function tokenOfOwnerByIndex(address _owner, uint256 _index) external view returns (uint256);
+}
+```
+
+### Minting Extension
+The **minting extension** is OPTIONAL for KIP-17 smart contracts. This allows your contract to mint a new token.
+Note: `KIP17MetadataMintable` is mutually exclusive with `KIP17Mintable`. If you want to use tokenURI, use `KIP17MetadataMintable`.
+
+```solidity
+pragma solidity 0.4.24;
+
+/// @title KIP-17 Non-Fungible Token Standard, optional minting extension
+///  Note: the KIP-13 identifier for this interface is 0xeab83e20.
+interface KIP17Mintable {
+    /// @notice Create a new token
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    /// @param _to The account that will receive the minted token
+    /// @param _tokenId The token ID to mint
+    /// @return True if the minting operation is successful, false otherwise
+    function mint(address _to, uint256 _tokenId) public returns (bool);
+
+    /// @notice Check the minting permission
+    /// @param _account The account to check the minting permission
+    /// @return True if the account has the minting permission, false otherwise
+    function isMinter(address _account) public view returns (bool);
+
+    /// @notice Give the minting permission to `_account`
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    /// @param _account The account to be given the minting permission
+    function addMinter(address _account) public;
+
+    /// @notice Renounce the minter permission of `msg.sender`
+    /// @dev Throws if `msg.sender` is not allowed to mint
+    function renounceMinter() public;
+}
+```
+
+### Minting with URI Extension
```

Comment:
It would be nice to add a comment on the relationship(? or dependency?) between metadataMintable and metadata.
( maybe "A metadata extension must be implemented to implement metadataMintable" like this? )

Reply:
Yup. I added. Please take a look. Thanks for the comment! :)